lib/network.cpp

```diff
@@ -168,9 +168,13 @@ int resolve_hostname(const char* hostname, sockaddr_storage &ip_addr) {
     hints.ai_protocol = IPPROTO_TCP;
     int retval = getaddrinfo(hostname, NULL, &hints, &res);
     if (retval) {
-        char buf[256];
-        snprintf(buf, sizeof(buf), "%s: getaddrinfo", time_to_string(dtime()));
-        perror(buf);
+        char buf[512];
+        snprintf(buf, sizeof(buf), "%s: getaddrinfo(%s) returned %s", time_to_string(dtime()), hostname, gai_strerror(retval));
```

Comment:
I personally don't like the "chaining" using ':' like this: "${date}: ${function}: ${error cause}: {system error}". For me the ':' have different meanings. The first means "happend at", the second "failed because", the third "this also belongs to this error". In writing this I realize that there is a better solution (although it may not be thread-safe). I'll change the PR shortly.